src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/CompiledPageRouteModelProvider.cs

```diff
@@ -75,7 +75,25 @@ public void OnProvidersExecuted(PageRouteModelProviderContext context)
             }
         }
 
-        private void CreateModels(IList<PageRouteModel> results)
+        /// <summary>
+        /// Gets the sequence of <see cref="CompiledViewDescriptor"/> from <paramref name="applicationManager"/>.
+        /// </summary>
+        /// <param name="applicationManager">The <see cref="ApplicationPartManager"/>s</param>
+        /// <returns>The sequence of <see cref="CompiledViewDescriptor"/>.</returns>
+        protected virtual IEnumerable<CompiledViewDescriptor> GetViewDescriptors(ApplicationPartManager applicationManager)
```

Comment:
Does this really need to be `protected virtual`? This kind of API defeats the reason why we use a context object.